### PR TITLE
Support encoding.Text(Unm|M)arshaler

### PR DESCRIPTION
**Pull request by kyoh86**
I'm fed up with implementing "MarhsalXXX", "Stringer", etc, etc...
kingpin requires "Value" interface to parse string
even though there's cool interface "encoding.TextUnmarshaler".

### CODE REVIEW

File: values.go [hunk 1]
```diff
@@ -42,23 +43,45 @@ type Getter interface {
...
+
+func (w wrapText) String() string {
+	buf, _ := w.text.MarshalText()
+	return string(buf)
+}
+
+func (w *wrapText) Set(s string) error {
+	return w.text.UnmarshalText([]byte(s))
+}
+
```
Comment (alecthomas) [inline values.go]: You'll need to implement `IsCumulative()` on `wrapText`, and conditionally forward it to `w.text`.

Comment (kyoh86) [reply values.go]: oh

Comment (kyoh86) [reply values.go]: :thinking: `UnmarshalText` is not implemented to parse one item, but to parse "whole" items.
i.e. `Value.Set` cumulates item, but `UnmarshalText` overwrite all of items.

The wrapText should not implement `IsCumulative`, should it?

Comment (kyoh86) [reply values.go]: I want to put `MarshalText` to use, but I have no idea.

Comment (alecthomas) [reply values.go]: Sorry, I meant the `IsBoolFlag()` method of the `boolFlag` interface. But I'm not 100% sure that's useful, so maybe don't worry about it.

**Comment by alecthomas** [comment]
BTW, the `Value` interface is adopted from the [stdlib's flag package](https://godoc.org/flag#Value), so it's not something specific to Kingpin :)

**Pull request merged**